docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md

```diff
@@ -101,33 +111,76 @@ function addItemAttachment(itemId) {
+To get an attachment's content, you can use the [getAttachmentContentAsync](/office/dev/add-ins/reference/objectmodel/preview-requirement-set/office.context.mailbox.item#getattachmentcontentasyncattachmentid-options-callback--attachmentcontent) method. The supported formats are listed in the [AttachmentContentFormat](/javascript/api/outlook/office.mailboxenums.attachmentcontentformat) enum.
+
+You should provide a callback method to check for the status and any error by using the `AsyncResult` output parameter object. You can also pass any additional parameters to the callback method by using the optional `asyncContext` parameter.
+
+The following JavaScript example gets the attachments and allows you to set up distinct handling for each supported attachment format.
+
+```js
+var item = Office.context.mailbox.item;
+var options = {asyncContext: {currentItem: item}};
+item.getAttachmentsAsync(options, callback);
+
+function callback(result) {
+  if (result.value.length > 0) {
+    for (i = 0 ; i < result.value.length ; i++) {
+      result.asyncContext.currentItem.getAttachmentContentAsync(result.value[i].id, handleAttachmentsCallback);
+    }
+  }
+}
+
+function handleAttachmentsCallback(result) {
+  // Parse string to be a url, an .eml file, a base64-encoded string, or an .icalendar file.
+  if (result.value.format === Office.MailboxEnums.AttachmentContentFormat.Base64) {
+    // Handle file attachment.
+  } else if (result.value.format === Office.MailboxEnums.AttachmentContentFormat.Eml) {
+    // Handle email item attachment.
+  } else if (result.value.format === Office.MailboxEnums.AttachmentContentFormat.ICalendar) {
+    // Handle .icalender attachment.
+  } else if (result.value.format === Office.MailboxEnums.AttachmentContentFormat.Url) {
+    // Handle cloud attachment.
+  } else {
+    // Handle attachment formats that are not supported.
+  }
+}
+```
+
+## Remove an attachment
+
+You can remove a file or item attachment from a message or appointment item in a compose form by specifying the corresponding attachment ID and using the [removeAttachmentAsync](/office/dev/add-ins/reference/objectmodel/requirement-set-1.5/Office.context.mailbox.item#removeattachmentasyncattachmentid-options-callback) method. You should only remove attachments that the same add-in has added in the same session. You should make sure the attachment ID corresponds to a valid attachment, or the method will return an error. Similar to the `addFileAttachmentAsync` and `addItemAttachmentAsync` methods, `removeAttachmentAsync` is an asynchronous method. You should provide a callback method to check for the status and any error by using the `AsyncResult` output parameter object. You can also pass any additional parameters to the callback method by using the optional `asyncContext` parameter.
```

Comment:
Do we really  need to say "You should make sure the attachment ID corresponds to a valid attachment, or the method will return an error."?